pkg/trace/api/otlp.go

```diff
@@ -187,6 +188,147 @@ func (o *OTLPReceiver) SetOTelAttributeTranslator(attrstrans *attributes.Transla
+	if !srcok {
+		if v, ok := resourceAttributesMap["_dd.hostname"]; ok {
+			src = source.Source{Kind: source.HostnameKind, Identifier: v}
+			srcok = true
+		}
+	}
+	// TODO(songy23): use AttributeDeploymentEnvironmentName once collector version upgrade is unblocked
+	env := traceutil.GetOTelAttrVal(resourceAttributes, true, "deployment.environment.name", semconv.AttributeDeploymentEnvironment)
+	lang := traceutil.GetOTelAttrVal(resourceAttributes, true, semconv.AttributeTelemetrySDKLanguage)
+	if lang == "" {
+		lang = fastHeaderGet(httpHeader, header.Lang)
+	}
+	containerID := traceutil.GetOTelAttrVal(resourceAttributes, true, semconv.AttributeContainerID, semconv.AttributeK8SPodUID)
+	if containerID == "" {
+		containerID = o.cidProvider.GetContainerID(ctx, httpHeader)
+	}
+
+	// 2. Transform OTLP spans to DD spans. If metadata was missing in resource attributes, attempt to get it from spans
+	topLevelByKind := o.conf.HasFeature("enable_otlp_compute_top_level_by_span_kind")
+	ignoreResNames := make(map[string]struct{})
+	for _, resName := range o.conf.Ignore["resource"] {
+		ignoreResNames[resName] = struct{}{}
+	}
+	tracesByID := make(map[uint64]pb.Trace)
+	priorityByID := make(map[uint64]sampler.SamplingPriority)
+	var spancount int64
+	for j := 0; j < rspans.ScopeSpans().Len(); j++ {
+		libspans := rspans.ScopeSpans().At(j)
+		for k := 0; k < libspans.Spans().Len(); k++ {
+			otelspan := libspans.Spans().At(k)
+			if _, exists := ignoreResNames[traceutil.GetOTelResource(otelspan, otelres)]; exists {
+				continue
+			}
+
+			spancount++
+			traceID := traceutil.OTelTraceIDToUint64(otelspan.TraceID())
+			if tracesByID[traceID] == nil {
+				tracesByID[traceID] = pb.Trace{}
+			}
+			ddspan := transform.OtelSpanToDDSpan(otelspan, otelres, libspans.Scope(), topLevelByKind, o.conf, nil)
+			if !srcok {
+				src, srcok = traceutil.GetOtelSource(otelspan, otelres, o.conf.OTLPReceiver.AttributesTranslator)
+			}
+			if env == "" {
+				// no env at resource level, try the first span
+				if v := ddspan.Meta["env"]; v != "" {
+					env = v
+				}
+			}
+			if containerID == "" {
+				// no cid at resource level, grab what we can
+				_, containerID = transform.GetFirstFromMap(ddspan.Meta, semconv.AttributeContainerID, semconv.AttributeK8SPodUID)
+			}
+
+			if p, ok := ddspan.Metrics["_sampling_priority_v1"]; ok {
+				priorityByID[traceID] = sampler.SamplingPriority(p)
+			}
+			tracesByID[traceID] = append(tracesByID[traceID], ddspan)
+		}
+	}
+
+	tagstats := &info.TagStats{
+		Tags: info.Tags{
+			Lang:            lang,
+			TracerVersion:   fmt.Sprintf("otlp-%s", resourceAttributesMap[semconv.AttributeTelemetrySDKVersion]),
+			EndpointVersion: "opentelemetry_grpc_v1",
+		},
+		Stats: info.NewStats(),
+	}
+
+	tags := tagstats.AsTags()
+	_ = o.statsd.Count("datadog.trace_agent.otlp.spans", spancount, tags, 1)
+	_ = o.statsd.Count("datadog.trace_agent.otlp.traces", int64(len(tracesByID)), tags, 1)
+	p := Payload{
```

Comment:
We created a hack to read from http headers . we should stop doing so https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/4a5e6db494acb2d56617266828f8db03957d3960/exporter/datadogexporter/traces_exporter.go#L134

Reply:
Are all the headers deprecated, including `header.ComputedStats`? Should I remove all of them in this code?